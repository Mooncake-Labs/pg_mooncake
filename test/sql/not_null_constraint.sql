CREATE TABLE t (c1 INT NOT NULL, c2 INT) using columnstore;

INSERT INTO t VALUES (1, 1);
INSERT INTO t (c1) VALUES (2);
SELECT * FROM t;

INSERT INTO t VALUES (NULL, 3);
INSERT INTO t (c1) VALUES (NULL);
SELECT * FROM t;

DROP TABLE t;
