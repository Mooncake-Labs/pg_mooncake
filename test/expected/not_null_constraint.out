CREATE TABLE t (c1 INT NOT NULL, c2 INT) using columnstore;
INSERT INTO t VALUES (1, 1);
INSERT INTO t (c1) VALUES (2);
SELECT * FROM t;
 c1 | c2 
----+----
  1 |  1
  2 |   
(2 rows)

INSERT INTO t VALUES (NULL, 3);
ERROR:  (PGDuckDB/Duckdb_ExecCustomScan) Constraint Error: NOT NULL constraint failed: t.c1
INSERT INTO t (c1) VALUES (NULL);
ERROR:  (PGDuckDB/Duckdb_ExecCustomScan) Constraint Error: NOT NULL constraint failed: t.c1
SELECT * FROM t;
 c1 | c2 
----+----
  1 |  1
  2 |   
(2 rows)

DROP TABLE t;
